libs/client/src/request/mod.rs

```diff
@@ -351,6 +431,32 @@ impl SolRpcRequest for GetSlotRequest {
     }
 }
 
+pub type GetSlotRequestBuilder<R> = RequestBuilder<
+    R,
+    GetSlotRpcConfig,
+    Option<GetSlotParams>,
+    MultiRpcResult<Slot>,
+    MultiRpcResult<Slot>,
+>;
+
+impl<R> GetSlotRequestBuilder<R> {
+    /// Change the `commitment` parameter for a `getSlot` request.
+    pub fn with_commitment(mut self, commitment_level: CommitmentLevel) -> Self {
+        let mut config = self.request.params.unwrap_or_default();
+        config.commitment = Some(commitment_level);
+        self.request.params = Some(config);
```

Comment:
nit: here and below, you can simplify this with [`get_or_insert_default`](https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert_default):
```suggestion
        let mut params = self.request.params.get_or_insert_default();
        config.commitment = Some(commitment_level);
```